src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInMetricsListener.cs

```diff
@@ -84,43 +109,87 @@ public override void OnEventWritten(string name, object payload)
-                tags.Add(new KeyValuePair<string, object>(SemanticConventions.AttributeHttpScheme, context.Request.Scheme));
-                tags.Add(new KeyValuePair<string, object>(SemanticConventions.AttributeHttpMethod, context.Request.Method));
-                tags.Add(new KeyValuePair<string, object>(SemanticConventions.AttributeHttpStatusCode, TelemetryHelper.GetBoxedStatusCode(context.Response.StatusCode)));
+                tags.Add(new KeyValuePair<string, object>(SemanticConventions.AttributeNetHostName, context.Request.Host.Host));
 
-                if (context.Request.Host.HasValue)
+                if (context.Request.Host.Port is not null && context.Request.Host.Port != 80 && context.Request.Host.Port != 443)
                 {
-                    tags.Add(new KeyValuePair<string, object>(SemanticConventions.AttributeNetHostName, context.Request.Host.Host));
+                    tags.Add(new KeyValuePair<string, object>(SemanticConventions.AttributeNetHostPort, context.Request.Host.Port));
+                }
+            }
 
-                    if (context.Request.Host.Port is not null && context.Request.Host.Port != 80 && context.Request.Host.Port != 443)
-                    {
-                        tags.Add(new KeyValuePair<string, object>(SemanticConventions.AttributeNetHostPort, context.Request.Host.Port));
-                    }
+#if NET6_0_OR_GREATER
+            var route = (context.GetEndpoint() as RouteEndpoint)?.RoutePattern.RawText;
+            if (!string.IsNullOrEmpty(route))
+            {
+                tags.Add(new KeyValuePair<string, object>(SemanticConventions.AttributeHttpRoute, route));
+            }
+#endif
+            if (this.options.Enrich != null)
+            {
+                try
+                {
+                    this.options.Enrich(HttpServerDurationMetricName, context, ref tags);
+                }
+                catch (Exception ex)
+                {
+                    AspNetCoreInstrumentationEventSource.Log.EnrichmentException(nameof(HttpInMetricsListener), EventName, HttpServerDurationMetricName, ex);
                 }
             }
 
-            // see the spec https://github.com/open-telemetry/semantic-conventions/blob/v1.21.0/docs/http/http-spans.md
-            if (this.emitNewAttributes)
+            // We are relying here on ASP.NET Core to set duration before writing the stop event.
+            // https://github.com/dotnet/aspnetcore/blob/d6fa351048617ae1c8b47493ba1abbe94c3a24cf/src/Hosting/Hosting/src/Internal/HostingApplicationDiagnostics.cs#L449
+            // TODO: Follow up with .NET team if we can continue to rely on this behavior.
+            this.httpServerDuration.Record(Activity.Current.Duration.TotalMilliseconds, tags);
+        }
+    }
+
+    public void OnEventWritten_New(string name, object payload)
+    {
+        if (name == OnStopEvent)
```

Comment:
nit: you can move this check to `OnEventWritten`

Reply:
done